io/subsystem/src/main/java/org/wildfly/extension/io/WorkerResourceDefinition.java

```diff
@@ -166,7 +166,10 @@ public void registerAttributes(ManagementResourceRegistration resourceRegistrati
                 new WorkerWriteAttributeHandler(WORKER_TASK_MAX_THREADS){
                     @Override
                     boolean setValue(XnioWorker worker, ModelNode value) throws IOException {
-                        return worker.setOption(Options.WORKER_TASK_MAX_THREADS, value.asInt()) == null;
+                        if (value.isDefined()) {
```

Comment:
Thanks for this. Even if we decide that these various WorkerWriteAttributeHandler subclasses should return false even if worker.setOption returns non-null, if !value.isDefined() the setOption shouldn't be called and the method should return true.